tests/manual/media/FXMediaPlayer/src/fxmediaplayer/media/FXMedia.java

```diff
@@ -0,0 +1,141 @@
+ * [macOS/Linux] ./dist/FXMediaPlayer/bin/FXMediaPlayer
+ * [Windows] jlink --output dist/FXMediaPlayer -p ../../../../build/jmods;dist --add-modules javafx.base,javafx.controls,javafx.media,javafx.graphics,FXMediaPlayer --launcher FXMediaPlayer=FXMediaPlayer/fxmediaplayer.FXMediaPlayer
+ * [Windows] dist\FXMediaPlayer\bin\FXMediaPlayer.bat
+ */
+public class FXMedia {
+
+    private static List<String> SUPPORTED_EXT = new ArrayList<>();
+    private static boolean isJRT = true;
+
+    static {
+        SUPPORTED_EXT.add("mp3");
+        SUPPORTED_EXT.add("wav");
+        SUPPORTED_EXT.add("aif");
+        SUPPORTED_EXT.add("mp4");
+        SUPPORTED_EXT.add("m4a");
+        SUPPORTED_EXT.add("m4v");
+    }
+
+    public static List<String> getEmbededMediaFiles() {
+        List<String> sources = new ArrayList<>();
+
+        try {
+            Path path = null;
+            URI uri = null;
+            Stream<Path> walk = null;
+            try {
+                FileSystem fileSystem = FileSystems.getFileSystem(URI.create("jrt:/"));
+                path = fileSystem.getPath("modules", "FXMediaPlayer",
+                        "fxmediaplayer", "media");
+                walk = Files.walk(path, 1);
+            } catch (NoSuchFileException ex) {
+                isJRT = false;
+                uri = FXMedia.class.getResource("/fxmediaplayer/media").toURI();
+
+                if (uri.getScheme().equals("jar")) {
+                    FileSystem fileSystem = FileSystems.newFileSystem(uri,
+                            Collections.<String, Object>emptyMap());
+                    path = fileSystem.getPath("fxmediaplayer", "media");
+                } else {
+                    path = Path.of(uri);
+                }
+                walk = Files.walk(path, 1);
+            }
+
+            Predicate<Path> predicate = new Predicate<Path>() {
+                @Override
+                public boolean test(Path p) {
+                    final AtomicBoolean isSupported = new AtomicBoolean(false);
+                    SUPPORTED_EXT.stream().forEach(ext -> {
+                        if (p.getFileName().toString().endsWith(ext)) {
+                            isSupported.set(true);
+                        }
+                    });
+
+                    return isSupported.get();
+                }
+            };
+
+            final URI uri2 = uri;
+            walk.filter(predicate).forEach(p -> {
+                if (isJRT()) {
+                    sources.add("jrt:///FXMediaPlayer/fxmediaplayer/media" +
+                        "/" + p.getFileName());
+                } else {
+                    sources.add(uri2.toString() + "/" + p.getFileName());
+                }
+            });
+        } catch (URISyntaxException | IOException ex) {
+            System.out.println("Exception: " + ex);
```

Comment:
Since this is a test app, I wouldn't bother with logging. We generally would use `System.err` rather than `System.out` for printing exceptions, but that's a minor point.

Reply:
I change it to System.err.